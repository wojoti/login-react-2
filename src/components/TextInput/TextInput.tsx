import React from "react";
import { Wrapper } from "./TextInput.style";

type Props = {
  type: string;
  name?: string;
  id?: string;
  onFieldChange?: (e: any) => void;
};

const TextInput = (props: Props) => {
  return (
    <Wrapper
      type={props.type}
      name={props.name}
      id={props.id}
      onChange={props.onFieldChange}
    ></Wrapper>
  );
};

export default TextInput;
