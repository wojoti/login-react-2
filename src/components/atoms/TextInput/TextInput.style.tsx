import styled from 'styled-components'
interface Props {}

export const StyledTextInput = styled.input<Props>`
  width: 100%;
  height: 32px;
  border: 2px solid #e5e7eb;
  border-radius: 5px;
  padding-left: 10px;
`
