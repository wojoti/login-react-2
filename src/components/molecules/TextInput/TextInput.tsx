import Label from "@components/atoms/Label/Label";
import TextField from "@components/atoms/TextField/TextField";
import { forwardRef } from "react";
import Wrapper from "./TextInput.style";

export interface TextInputProps {
  type: string;
  name: string;
  id: string;
  onChange?: React.ChangeEventHandler<HTMLInputElement>;
  isValid?: boolean;
  testId?: string;
}
export type TextInputRef = HTMLInputElement;

const TextInput = forwardRef<TextInputRef, TextInputProps>(
  ({ type, name, id, onChange, isValid, testId }, ref) => (
    <Wrapper data-testid={testId}>
      <Label testId="textinput-label">{name}</Label>
      <TextField
        type={type}
        name={name}
        id={id}
        onChange={onChange}
        isValid={isValid}
        testId="textinput-textfield"
        ref={ref}
      />
    </Wrapper>
  )
);
TextInput.displayName = "TextInput";

export default TextInput;
