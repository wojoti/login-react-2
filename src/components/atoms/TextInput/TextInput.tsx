import React from 'react'
import { StyledTextInput } from './TextInput.style'

interface Props {
  type: string
  name: string
  id: string
  onFieldChange?: React.ChangeEventHandler<HTMLInputElement>
}

const TextInput = (props: Props) => {
  return (
    <StyledTextInput
      type={props.type}
      name={props.name}
      id={props.id}
      onChange={props.onFieldChange}
    ></StyledTextInput>
  )
}

export default TextInput
