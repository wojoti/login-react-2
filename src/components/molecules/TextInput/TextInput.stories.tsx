import { ComponentMeta, ComponentStory } from "@storybook/react";

import TextInput from "./TextInput";

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: "Molecules/TextInput",
  component: TextInput,
} as ComponentMeta<typeof TextInput>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof TextInput> = (args) => (
  <TextInput {...args} />
);

export const ValidEmail = Template.bind({});
ValidEmail.args = { type: "email", name: "email", id: "email", isValid: true };
export const InvalidEmail = Template.bind({});
InvalidEmail.args = {
  type: "email",
  name: "email",
  id: "email",
  isValid: false,
};

export const Password = Template.bind({});
Password.args = { type: "password", name: "password", id: "password" };
