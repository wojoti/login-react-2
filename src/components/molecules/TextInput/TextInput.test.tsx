import { render, screen, within } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import TextInput, { TextInputProps } from "./TextInput";

const props: TextInputProps = {
  type: "email",
  name: "test textinput",
  id: "id-textinput",
  testId: "test-textinput-id",
};

test("should match snapshot", () => {
  const tree = TestRenderer.create(<TextInput {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render textinput", () => {
  render(<TextInput {...props} />);
  const textinputElement = screen.getByTestId("test-textinput-id");
  expect(textinputElement).toBeInTheDocument();
});

test("should render textinput with certain label text", () => {
  render(<TextInput {...props} />);
  const textinputElement = screen.getByTestId("test-textinput-id");
  expect(textinputElement).toBeInTheDocument();
  const textinputLabelElement =
    within(textinputElement).getByTestId("textinput-label");
  expect(textinputLabelElement).toBeInTheDocument();
  expect(textinputLabelElement).toHaveTextContent("test textinput");
});

test("should render textinput with certain textfield id", () => {
  render(<TextInput {...props} />);
  const textinputElement = screen.getByTestId("test-textinput-id");
  expect(textinputElement).toBeInTheDocument();
  const textinputTextFieldElement = within(textinputElement).getByTestId(
    "textinput-textfield"
  );
  expect(textinputTextFieldElement).toBeInTheDocument();
  expect(textinputTextFieldElement).toHaveAttribute("id", "id-textinput");
});
