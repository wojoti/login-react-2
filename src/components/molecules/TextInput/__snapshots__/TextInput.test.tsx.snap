// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.c1 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: inherit;
  font-size: 14px;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 100%;
  height: 32px;
  border: 2px solid #e5e7eb;
  border-radius: 5px;
  padding-left: 10px;
}

.c0 {
  width: 100%;
}

<div
  className="c0"
  data-testid="test-textinput-id"
>
  <span
    className="c1"
    data-testid="textinput-label"
  >
    test textinput
  </span>
  <input
    className="c2"
    data-testid="textinput-textfield"
    id="id-textinput"
    name="test textinput"
    type="email"
  />
</div>
`;
